tools/upload_vhd.sh

```diff
@@ -0,0 +1,98 @@
+#!/bin/bash
+set -exuo pipefail
+
+BP=bootc-vhd
+
+case "$TEST_OS" in
+"rhel-9-4")
+	BASEURL="http://${DOWNLOAD_NODE}/rhel-9/nightly/RHEL-9/latest-RHEL-9.4.0"
+	;;
+"centos-stream-9")
+	BASEURL="https://composes.stream.centos.org/production/latest-CentOS-Stream"
+	;;
+*)
+	redprint "Variable TEST_OS has to be defined"
+	exit 1
+	;;
+esac
+
+dnf install -y osbuild-composer composer-cli cockpit-composer
+systemctl enable --now osbuild-composer.socket
+systemctl enable --now cockpit.socket
+systemctl restart osbuild-composer
+composer-cli status show
+mkdir -p /etc/osbuild-composer/repositories
+tee /etc/osbuild-composer/repositories/$(echo "$TEST_OS" | sed 's/\(.*\)-\(.*\)/\1.\2/').json >/dev/null <<EOF
+{
+     "x86_64": [
+        {
+            "name": "baseos",
+            "baseurl": "$BASEURL/compose/BaseOS/x86_64/os/",
+            "check_gpg": false,
+            "rhsm": false
+        },
+        {
+            "name": "appstream",
+            "baseurl": "$BASEURL/compose/AppStream/x86_64/os/",
+            "check_gpg": false,
+            "rhsm": false
+        }
+    ]
+}
+EOF
+tee bp.toml >/dev/null <<EOF
+name = "$BP"
+version = "0.0.1"
+
+packages = [
+    { name = "cloud-init" },
+    { name = "WALinuxAgent" },
+    { name = "podman" },
```

Comment:
Interesting...wait, why are we generating a custom azure image?  The goal is to test installing via os-replace on the stock RHEL one, right?

Reply:
Right. But there's no RHEL9 and CS9 image on Azure. We have to upload by our own and use it to boot first. Then run `bootc install`.

Reply:
Wait...really?  Wow, OK yep https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html/deploying_rhel_9_on_microsoft_azure/index is quite clear.

Though...if this is the case, in the end I'm not actually sure that this is as important for us to test, because this use case from a customer PoV I think basically becomes using bootc-image-builder to upload the target container directly to Azure, right?  Indirecting in this way would feel strange.

Not saying there's anything wrong with this test though!  We might as well do it, so don't block merging.  I actually love how it's combining things end to end (nice work!).

Reply:
Update: https://learn.microsoft.com/en-us/azure/virtual-machines/workloads/redhat/redhat-images#rhel-9-image-types

Reply:
Thanks! That's not free, and needs rhc subscription.